docs/Developer-Guide_Build-Preparation.md

```diff
@@ -42,8 +42,10 @@ Make sure that full path to the build script **does not contain spaces**.
 
 ## Providing build configuration
 
-After the first run of `compile.sh` a new configuration file `config-example.conf` and symlink `config-default.conf` will be created.
-You may edit it to your needs or create different configuration files using it as a template.
+After the first run of `compile.sh` a new configuration file `config-example.conf` and symlink `config-default.conf` will be created in `userpatches`.
+You may edit it to your needs or create different configuration files using it as a template. To start a build with a specific configuration file, e.g. for `config-foo.conf`:
```

Comment:
Configurations are supported but one needs to make it on their own. Its just a collection of parameters.

This should be reworked, previous statement removed.

Reply:
_A configuration file named `userpatches/config-<something>.conf.sh` (`.conf` also allowed) is a bash script that is sourced during the build if you use `./compile.sh something`. You can use it to pass the parameters you'd normally  pass on the command line (`PARAM1=value1 PARAM2=value`) by using the same syntax, but put each PARAM on a separate line. Command-line parameters still are able to override what is the config file. More advanced use cases can use conditionals, define functions to implement hooks, source other/common config files, etc. A few quite complex examples can be found in https://github.com/lanefu/armbian-userpatches-example-indiedroid-nova_